src/mono/IosAppBuilderTasks/Xcode.cs

```diff
@@ -0,0 +1,125 @@
+            // these libraries are pinvoked
+            toLink += $"    \"-force_load {lib}\"{Environment.NewLine}";
+        }
+        foreach (string lib in Directory.GetFiles(binDir, "*.dll.o"))
+        {
+            // these libraries are linked via modules.m
+            toLink += $"    \"{lib}\"{Environment.NewLine}";
+        }
+        cmakeLists = cmakeLists.Replace("%NativeLibrariesToLink%", toLink);
+
+        string plist = Utils.GetEmbeddedResource("Info.plist.template")
+            .Replace("%BundleIdentifier%", projectName);
+
+        File.WriteAllText(Path.Combine(binDir, "Info.plist.in"), plist);
+        File.WriteAllText(Path.Combine(binDir, "CMakeLists.txt"), cmakeLists);
+
+        var cmakeArgs = new StringBuilder();
+        cmakeArgs
+            .Append("-S.")
+            .Append(" -B").Append(projectName)
+            .Append(" -GXcode")
+            .Append(" -DCMAKE_SYSTEM_NAME=iOS")
+            .Append(" \"-DCMAKE_OSX_ARCHITECTURES=arm64;x86_64\"")
+            .Append(" -DCMAKE_OSX_DEPLOYMENT_TARGET=10.1")
+            .Append(" -DCMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH=NO");
+
+        File.WriteAllText(Path.Combine(binDir, "runtime.h"),
+            Utils.GetEmbeddedResource("runtime.h"));
+
+        // forward pinvokes to "__Internal"
+        string dllMap = string.Join(Environment.NewLine, Directory.GetFiles(workspace, "*.a")
+            .Select(f => $"    mono_dllmap_insert (NULL, \"{Path.GetFileNameWithoutExtension(f)}\", NULL, \"__Internal\", NULL);"));
+
+        File.WriteAllText(Path.Combine(binDir, "runtime.m"),
+            Utils.GetEmbeddedResource("runtime.m")
+                .Replace("//%DllMap%", dllMap)
+                .Replace("%EntryPointLibName%", Path.GetFileName(entryPointLib)));
+
+        Utils.RunProcess("cmake", cmakeArgs.ToString(), workingDir: binDir);
+
+        return Path.Combine(binDir, projectName, projectName + ".xcodeproj");
+    }
+
+    public static string BuildAppBundle(
+        string xcodePrjPath, string architecture, bool optimized, string? devTeamProvisioning = null)
+    {
+        string sdk = "";
+        var args = new StringBuilder();
+        args.Append("ONLY_ACTIVE_ARCH=NO");
+        if (architecture == "arm64")
+        {
+            sdk = "iphoneos";
+            args.Append(" -arch arm64")
+                .Append(" -sdk iphoneos")
+                .Append(" -allowProvisioningUpdates")
+                .Append(" DEVELOPMENT_TEAM=").Append(devTeamProvisioning);
+        }
+        else
+        {
+            sdk = "iphonesimulator";
+            args.Append(" -arch x86_64")
+                .Append(" -sdk iphonesimulator");
+        }
+
+        string config = optimized ? "Release" : "Debug";
+        args.Append(" -configuration ").Append(config);
+
+        Utils.RunProcess("xcodebuild", args.ToString(), workingDir: Path.GetDirectoryName(xcodePrjPath));
+        return Path.Combine(Path.GetDirectoryName(xcodePrjPath)!, config + "-" + sdk,
```

Comment:
Not sure if we should hard code the path here. I think it might be more consistent to store the output to artefacts folder with RID used as sub-folder